test/token/ERC20/extensions/ERC4626.test.js

```diff
@@ -197,7 +197,7 @@ contract('ERC4626', function (accounts) {
       );
 
       // Price before
-      const sharesBefore = await vault.previewWithdraw(amount);
+      const assetsBefore = await vault.previewWithdraw(amount);
```

Comment:
`Withdraw` (and `previewWithdraw`) take assets as input, and output shares.

In the case of preview, its equivalent to "I want to get X underlying token, how many shares do I need to burn". So this is NOT `assetsBefore`

Also in lines 210 and 211 you are confusing the two.

It happens to works because the rate is 1:1, but it won't it the rates are modified (for example if we run this tests with an offset)